pkg/services/dashboards/dashboard_service.go

```diff
@@ -129,6 +134,23 @@ func (dr *dashboardServiceImpl) buildSaveDashboardCommand(dto *SaveDashboardDTO,
 	return cmd, nil
 }
 
+func (dr *dashboardServiceImpl) validateDashboardIsNotProvisioned(dashboardId int64) error {
+	dpQuery := &models.GetProvisionedDashboardByDashboardId{DashboardId: dashboardId}
+	err := bus.Dispatch(dpQuery)
+
+	// provisioned dashboards cannot be saved. So we can only save
+	// this dashboard if ErrDashboardProvisioningDoesNotExist is returned
+	if err != nil && err != models.ErrDashboardProvisioningDoesNotExist {
+		return err
+	}
+
+	if err == nil {
+		return models.ErrDashboardCannotSaveProvisionedDashboard
```

Comment:
Logic here would probably be easier to read if GetProvisionedDashboardByDashboardId handler returns err for actual error and set the cmd.Result. Then cmd.Result can be used to decide if error message should be returned or not.